source/common/umf_pools/disjoint_pool_config_parser.cpp

```diff
@@ -25,45 +34,81 @@ constexpr auto operator""_GB(unsigned long long x) -> size_t {
     return x * 1024 * 1024 * 1024;
 }
 
+DisjointPoolAllConfigs::~DisjointPoolAllConfigs() {
+    for (auto &Config : Configs) {
+        umfDisjointPoolParamsDestroy(Config);
+    }
+}
+
 DisjointPoolAllConfigs::DisjointPoolAllConfigs(int trace) {
+    umf_result_t ret = UMF_RESULT_SUCCESS;
+
     for (auto &Config : Configs) {
-        Config = umfDisjointPoolParamsDefault();
-        Config.PoolTrace = trace;
+        umf_disjoint_pool_params_handle_t params = NULL;
+        ret = umfDisjointPoolParamsCreate(&params);
+        if (ret != UMF_RESULT_SUCCESS) {
+            logger::error("DisjointPool params create failed");
+            return;
+        }
+        ret = umfDisjointPoolParamsSetTrace(params, trace);
+        CheckConfigRet(ret);
+
+        Config = params;
     }
 
-    Configs[DisjointPoolMemType::Host].Name = "Host";
-    Configs[DisjointPoolMemType::Device].Name = "Device";
-    Configs[DisjointPoolMemType::Shared].Name = "Shared";
-    Configs[DisjointPoolMemType::SharedReadOnly].Name = "SharedReadOnly";
+    ret = umfDisjointPoolParamsSetName(Configs[DisjointPoolMemType::Host], "Host");
```

Comment:
nit: you can just do `CheckConfigRet(umfDisjointPoolParamsSetName(Configs[DisjointPoolMemType::Host], "Host"))`.

Also, I would rename CheckConfigRet to `UMF_CALL` or something like that so it's shorter and more consistent with UR_CALL/ZE_CALL

Reply:
Done